gcc/rust/ast/rust-ast-dump.h

```diff
@@ -32,6 +32,8 @@ class Dump
 {
 public:
   Dump (std::ostream &stream);
+  Dump (std::ostream &stream, bool print_internal,
+	std::vector<std::string> blacklist);
```

Comment:
we can use a `std::set<std::string>` here for faster lookup, and for slightly better APIs as well since we can do `blacklist.contains(...)`. also, I would call the set `forbidden` or `excluded` instead of `blacklist`